### PR TITLE
Changes to allow setting of MTU size and jumbo frames.

### PR DESCRIPTION
**Pull request by mmnelemane**
With this change the environment variable "want_mtu_size" can be used to set desired MTU size for Admin and Storage nodes. Setting want_mtu_size value to "jumbo" makes it to take the maximum MTU size feasible (8900) for these nodes. The qa_crowbarsetup.sh is modified to change the 'network' proposal to set the MTU value.

### CODE REVIEW

File: scripts/mkcloud [hunk 1]
```diff
@@ -1083,6 +1083,9 @@ Optional
     want_docker=1 (default='')
         Deploy docker image (upload image to glance, create instance based on it etc.).
         Only possible with SLE12 node.
+    want_mtu_size=<size>!"jumbo" (default='')
```
Comment (jdsn) [inline scripts/mkcloud]: should be '|' instead of '!' - right?

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -49,6 +49,14 @@ export cinder_netapp_login
 export cinder_netapp_password
 export localreposdir_target
 export want_ipmi=${want_ipmi:-false}
+export want_mtu_size=${want_mtu_size:-1500}
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: this seems redundant, as the default already is 1500
so this line can be dropped

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -1211,6 +1219,21 @@ EOF
...
 
+    # Setup network attributes for JUMBO Frames.
+    if [ -n "${want_mtu_size##+([[:space:]])}" ]; then
+        if [ "$want_mtu_size" == "jumbo" ]; then
+            echo "Value of MTU set in admin and storage nodes is $jumbo_frame_size"
+            /opt/dell/bin/json-edit -a attributes.network.networks.storage.mtu -r -v $jumbo_frame_size $netfile
+            /opt/dell/bin/json-edit -a attributes.network.networks.admin.mtu -r -v $jumbo_frame_size $netfile
+        else
+            echo "Value of MTU set in admin and storage nodes is $want_mtu_size"
+            /opt/dell/bin/json-edit -a attributes.network.networks.storage.mtu -r -v $want_mtu_size $netfile
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: this is code duplication, I propose to optimise it to:
(assuming the line above gets dropped)
```
if [[ $want_mtu_size == "jumbo" ]] ; then
    want_mtu_size=8900
fi
if [[ $want_mtu_size ]] ; then
         echo "Setting MTU to custom value of: $want_mtu_size"
         /opt/dell/bin/json-edit -a attributes.network.networks.storage.mtu -r -v $jumbo_frame_size $netfile
         /opt/dell/bin/json-edit -a attributes.network.networks.admin.mtu -r -v $jumbo_frame_size $netfile
fi
```

File: scripts/qa_crowbarsetup.sh [hunk 4]
```diff
@@ -49,6 +49,14 @@ export cinder_netapp_login
 export cinder_netapp_password
 export localreposdir_target
 export want_ipmi=${want_ipmi:-false}
+export want_mtu_size=${want_mtu_size:-1500}
+
+# MTU for Jumbo frame is set to 8900 instead of 9000 as expected. This is 
+# because, in case of tunnelling, GRE adds 24 Bytes and VXLAN adds 50 bytes.
+# Allowing a buffer of 100 bytes ensures that tests carry out without 
+# causing forced fragmentation. 
+export jumbo_frame_size=${jumbo_frame_size:-8900}
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: with the new block below this line can be dropped as well

File: scripts/qa_crowbarsetup.sh [hunk 5]
```diff
@@ -49,6 +49,7 @@ export cinder_netapp_login
 export cinder_netapp_password
 export localreposdir_target
 export want_ipmi=${want_ipmi:-false}
+
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: unrelated new line change :)

**Comment by jdsn** [comment]
:+1: lgtm

**Comment by MaximilianMeister** [comment]
+1 (not sure if we need the `--raw` attribute but i think this is minor)

**Comment by bmwiedemann** [comment]
please rebase to current master and use the new onadmin_help function
